### PR TITLE
Remove crud from boot.iso

### PR DESCRIPTION
**Pull request by dashea**
Remove a bunch of stuff that webkit and *-firmware pulled in. This adds a --except option to the installpkg command to filter packages from globs before they are handed to dnf.

### CODE REVIEW

**Comment by dashea** [comment]
Added one more commit after realizing that the gnome icon theme doesn't really do anything anymore. That shaves another 10MB or so off of boot.iso.

File: share/templates.d/99-generic/runtime-install.tmpl [hunk 1]
```diff
@@ -20,7 +20,13 @@ installpkg pigz
 installpkg kernel kernel-modules kernel-modules-extra
 installpkg grubby
 %if basearch != "s390x":
-    installpkg *-firmware
+    ## skip the firmware for sound, video, and scanners, none of which will
+    ## do much good for the installer. Also skip uhd-firmware which is not
+    ## even a kernel firmware package.
+    installpkg *-firmware --except alsa* --except midisport-firmware \
+                          --except crystalhd-firmware --except ivtv-firmware \
```
Comment (bcl) [inline share/templates.d/99-generic/runtime-install.tmpl]: the pkgglob needs to go at the end. Or change what you wrote in ltmpl :)
Good idea though!

**Comment by dashea** [comment]
Redid the comment in ltmpl, since I liked the order I used in runtime-instal better.

File: src/pylorax/ltmpl.py [hunk 2]
```diff
@@ -503,9 +503,35 @@ def installpkg(self, *pkgs):
...
 
+        excludes = []
+        while '--except' in pkgs:
+            idx = pkgs.index('--except')
+            if len(pkgs) == idx+1:
+                raise ValueError("installpkg needs an argument after --except")
+
+            excludes.append(pkgs[idx+1])
+            pkgs = pkgs[:idx] + pkgs[idx+2:]
+
```
Comment (bcl) [inline src/pylorax/ltmpl.py]: This block expects the --except entries to all be at the start.

File: share/templates.d/99-generic/runtime-install.tmpl (hunk 1 above)
Comment (nullr0ute) [inline share/templates.d/99-generic/runtime-install.tmpl]: You could exclude iscan-firmware too (if you don't already)

File: src/pylorax/ltmpl.py (hunk 2 above)
Comment (dashea) [inline src/pylorax/ltmpl.py]: It looks for the --except entries anywhere in the list and pulls them out into the excludes list. So if you wanted you could interleave pkgglobs and --excepts.

File: share/templates.d/99-generic/runtime-install.tmpl (hunk 1 above)
Comment (dashea) [inline share/templates.d/99-generic/runtime-install.tmpl]: Yes, iscan-firmware is the next line down ;-) It was another that was already in runtime-cleanup that I moved to the install command.

File: src/pylorax/ltmpl.py (hunk 2 above)
Comment (bcl) [inline src/pylorax/ltmpl.py]: Oh geez, I was thinking of it like a for loop. ACK.

**Comment by dashea** [comment]
Pushed.

**Pull request closed without merging**